Microsoft.Azure.Cosmos/src/Routing/AvailabilityStrategy/CrossRegionParallelHedgingAvailabilityStrategy.cs

```diff
@@ -0,0 +1,541 @@
+            this.ThresholdStep = thresholdStep ?? TimeSpan.MaxValue;
+        }
+
+        internal override bool Enabled()
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// This method determines if the request should be sent with a parallel hedging availability strategy.
+        /// This availability strategy can only be used if the request is a read-only request on a document request.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns>whether the request should be a parallel hedging request.</returns>
+        internal bool ShouldHedge(RequestMessage request)
+        {
+            //Only use availability strategy for document point operations
+            if (request.ResourceType != ResourceType.Document)
+            {
+                return false;
+            }
+
+            //check to see if it is a not a read-only request
+            if (!OperationTypeExtensions.IsReadOperation(request.OperationType))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Execute the parallel hedging availability strategy
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="client"></param>
+        /// <param name="request"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>The response after executing cross region hedging</returns>
+        public override async Task<ResponseMessage> ExecuteAvailablityStrategyAsync(
+            Func<RequestMessage, CancellationToken, Task<ResponseMessage>> sender,
+            CosmosClient client,
+            RequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            if (!this.ShouldHedge(request))
+            {
+                return await sender(request, cancellationToken);
+            }
+
+            using (CancellationTokenSource cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
+            {
+                IReadOnlyCollection<string> availableRegions = client.DocumentClient.GlobalEndpointManager.GetAvailableReadEndpointsByLocation().Keys;
+                int i = 0;
+                List<Task<(bool, ResponseMessage)>> requestTasks = new List<Task<(bool, ResponseMessage)>>(availableRegions.Count);
+
+                Task<(bool, ResponseMessage)> primaryRequest = this.RequestSenderAndResultCheckAsync(
+                    sender,
+                    request,
+                    cancellationToken,
+                    cancellationTokenSource);
+
+                requestTasks.Add(primaryRequest);
+
+                Task<Task<(bool, ResponseMessage)>> allRequests;
+
+                ResponseMessage responseMessage;
+                
+                //available regions or hedge regions?
+                //Send out hedged requests
+                foreach (string region in availableRegions)
```

Comment:
```suggestion
                foreach (string region in availableRegions.skip(1))
```

And remove the if condition below.